packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -448,7 +414,12 @@ const VirtualListBaseFactory = (type) => {
 				newFirstIndex = this.calculateFirstIndex(props, wasFirstIndexMax, dataSizeDiff);
 			}
 
-			this.setState({firstIndex: newFirstIndex, numOfItems});
+			if (slient) {
+				this.state.firstIndex = newFirstIndex; // eslint-disable-line react/no-direct-mutation-state
```

Comment:
I'll try to find a better way.

Reply:
<On Feb 18th>
The code to change the state directly in `calculateMetrics` was removed.
`getStatesAndUpdateBounds` ( which name was `updateStateAndBounds` before ) returns a new state instead of updating the state directly.